beacon-chain/forkchoice/doubly-linked-tree/forkchoice.go

```diff
@@ -199,11 +203,20 @@ func (f *ForkChoice) updateCheckpoints(ctx context.Context, jc, fc *ethpb.Checkp
 					f.store.prevJustifiedCheckpoint = f.store.justifiedCheckpoint
 					f.store.justifiedCheckpoint = &forkchoicetypes.Checkpoint{Epoch: jc.Epoch,
 						Root: jcRoot}
+					if err := f.updateJustifiedBalances(ctx, jcRoot); err != nil {
+						f.store.checkpointsLock.Unlock()
```

Comment:
It feels better to just defer unlock this from the top

Reply:
It can't be deferred cause there's a release that it's needed in one of the branches